ldms/src/store/hello_stream/l2_stream_csv_store.c

```diff
@@ -0,0 +1,1165 @@
+		       stream_handle->stream);
+		rc = EPERM;
+		goto out;
+	}
+
+	/** msg will be populated. if the type was json,
+            entity will also be populated. */
+	if (stream_type == LDMSD_STREAM_STRING){
+		fprintf(stream_handle->file, "%s\n", msg);
+		if (!buffer){
+			fflush(stream_handle->file);
+			fsync(fileno(stream_handle->file));
+		}
+	} else if (stream_type == LDMSD_STREAM_JSON){
+		if (!e){
+			msglog(LDMSD_LERROR, PNAME ": why is entity NULL?\n");
+			rc = EINVAL;
+			goto out;
+		}
+
+		if (e->type != JSON_DICT_VALUE) {
+			msglog(LDMSD_LERROR,
+                               PNAME ": Expected a dict object, not a %s.\n",
+			       json_type_name(e->type));
+			rc = EINVAL;
+			goto out;
+		}
+
+                if (!stream_handle->dataline.header){
+                        rc = _get_header_from_data(&stream_handle->dataline,e);
+			if (rc != 0) {
+				msglog(LDMSD_LDEBUG,
+                                       PNAME ": error processing header from data <%d>\n", rc);
+				goto out;
+			}
+                        _print_header(stream_handle);
+		}
+
+		_print_data_lines(stream_handle, e);
+		if (!buffer){
+			fflush(stream_handle->file);
+			fsync(fileno(stream_handle->file));
+		}
+	} else {
+		msglog(LDMSD_LERROR, PNAME ": unknown stream type\n");
+		rc = EINVAL;
+		goto out;
+	}
+
+out:
+
+	pthread_mutex_unlock(&stream_handle->lock);
+	return rc;
+}
+
+
+static int open_streamstore(char* stream){
+        struct csv_stream_handle *stream_handle;
+        char *tmp_filename;
+        char *tmp_basename;
+	char *dpath;
+        FILE *tmp_file;
+        unsigned long tspath;
+        int rc = 0;
+
+
+	// already have the cfg_lock
+	if (!root_path || !container || !stream) {
+	     msglog(LDMSD_LERROR, PNAME ": config not called. cannot open.\n");
+	     return ENOENT;
+	}
+
+        //NOTE: unlike store, this doesn't have the possibility of closing yet.
+        stream_handle = idx_find(stream_idx, (void *)stream, strlen(stream));
+        if (stream_handle){
+                msglog(LDMSD_LERROR,
+                       PNAME ": cannot open stream item. already have it\n");
+		return EINVAL;
+	}
+
+	// add additional 12 for the timestamp
+	size_t pathlen =
+                strlen(root_path) + strlen(stream) + strlen(container) + 8 + 12;
+        dpath = malloc(pathlen);
+	tmp_basename = malloc(pathlen);
+        tmp_filename = malloc(pathlen);
+	if (!dpath || !tmp_filename || !tmp_basename){
+		rc = ENOMEM;
+		goto out;
+	}
+
+        tspath = (unsigned long)(time(NULL));
+        sprintf(dpath, "%s/%s", root_path, container);
+	sprintf(tmp_basename, "%s/%s/%s", root_path, container, stream);
+        sprintf(tmp_filename, "%s/%s/%s.%lu",
+                root_path, container, stream, tspath);
+
+	msglog(LDMSD_LDEBUG, PNAME ": stream '%s' will have file '%s'\n",
+               stream, tmp_filename);
+
+	/* create path if not already there. */
+	rc = mkdir(dpath, 0777);
```

Comment:
use instead here create_outdir from store_csv_common.h or at least  f_mkdir_p from ovis_util/util.h

Reply:
done. going with 0777 since that is what is in store_flatfile